ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -774,21 +785,32 @@ def _useRealization(self, info, rlz):
                                               objVal = self._objectiveVar
                                               )
 
-      # 2 @ n: Crossover from set of parents
-      # Create childrenCoordinates (x1,...,xM)
+    # 2 @ n: Crossover from set of parents
+    # Create childrenCoordinates (x1,...,xM)
+    # if crossover probability is a float, keep it as is. But, If it's a string, called appropriate function.
+      if(self._crossoverProbType == "static"):
+        crossOverProb = self._crossoverProb
+      if(self._crossoverProb == "linear"):
+        crossOverProb = getLinearCrossoverProbability(self.getIteration(traj),self.limit)
```

Comment:
What happens if the user enter's `Linear` instead of `linear`. when we spoke before I showed you that in the if statement you can add `.lower()` to convert the whole word to lower case. (e.g., `if(self._crossoverProb.lower() == "linear"):`). 
In addition, what if the user entered something else other than linear or quadratic, their should be a final `else` that raises an error. `self.raiseAnError(IOError, "{} is not implemeted!. Currently only 'linear' and 'quadratic' are implemented".format(self._crossoverProb))`